### PR TITLE
Pyton Web App with sign-in, calling MS Graph API, and sign-out

### PR DESCRIPTION
**Pull request by rayluo**
Per @abhidnya13 's request, "we should ... create (new) PRs if the refactor is too big". So we rolled back the latest 2 comments from PR #1, and create a separated PR here.

The main changes are:

* Remove the global MSAL app instance, so that we don't share same instance (and its cache!) among different end users.
* Refactor overall structure of flask controllers to reduce some transitional steps
* Views (i.e. html files) no longer need to embed "if logged_in" logic

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -122,37 +124,26 @@ In the steps below, "ClientID" is the same as "Application ID" or "AppId".
...
 ### Step 4: Run the sample
 
-- You will need to install MSAL Python library, Flask framework, Flask-Sessions for server side session management and requests using pip as follows:
+- You will need to install dependencies using pip as follows:
 ```Shell
-$ pip install msal
-$ pip install flask
-$ pip install Flask-Session
-$ pip install requests
+$ pip install -r requirements.txt
```
Comment (jmprieur) [inline README.md]: this is cool!

File: app_config.py [hunk 2]
```diff
@@ -1,5 +1,21 @@
-AUTHORITY = "https://login.microsoftonline.com/Enter_the_Tenant_Name_Here"
+import os
+
+# This pattern is defined in Flask's documentation here
+# https://flask.palletsprojects.com/en/1.1.x/config/#configuring-from-environment-variables
+CLIENT_SECRET = os.getenv("CLIENT_SECRET")
```
Comment (jmprieur) [inline app_config.py]: The app creation scripts still write in config.py, which is deleted ...
.... we don't have a way to automate these apps any longer.

Also , where would customers set the CLIENT_SECRET? on a post-it :) ? this makes life of customes wanting to test the sample a bit harder;

File: README.md [hunk 3]
```diff
@@ -50,7 +52,7 @@ git clone https://github.com/Azure-Samples/ms-identity-python-webapp.git
 
 or download and extract the repository .zip file.
 
-> Given that the name of the sample is quiet long, you might want to clone it in a folder close to the root of your hard drive, to avoid file size limitations on Windows.
+> Given that the name of the sample is quite long, you might want to clone it in a folder close to the root of your hard drive, to avoid file name lenghth limitations when running on Windows.
```
Comment (jmprieur) [inline README.md]: lenghth  => length

File: README.md [hunk 4]
```diff
@@ -122,37 +124,26 @@ In the steps below, "ClientID" is the same as "Application ID" or "AppId".
 
 #### Configure the pythonwebapp project
 
-> Note: if you used the setup scripts, the changes below will have been applied for you
+> Note: if you used the setup scripts, the changes below may have been applied for you
 
 1. Open the `app_config.py` file
 1. Find the app key `Enter_the_Tenant_Name_Here` and replace the existing value with your Azure AD tenant name.
-1. Find the app key `Enter_the_Client_Secret_Here` and replace the existing value with the key you saved during the creation of the `python-webapp` app, in the Azure portal.
+1. You could find the app key `Enter_the_Client_Secret_Here` and replace the existing value with the key you saved during the creation of the `python-webapp` app, in the Azure portal.
```
Comment (jmprieur) [inline README.md]: This whole section is no longer true. Set the environment variable?

File: app_config.py [hunk 5]
```diff
@@ -1,5 +1,25 @@
-AUTHORITY = "https://login.microsoftonline.com/Enter_the_Tenant_Name_Here"
+import os
+
+CLIENT_SECRET = "Enter_the_Client_Secret_Here" # Our Quickstart uses this placeholder
+# In your production app, we recommend you to use other ways to store your secret,
```
Comment (jmprieur) [inline app_config.py]: This looks good!

File: app_config.py [hunk 6]
```diff
@@ -1,5 +1,25 @@
...
+# AUTHORITY = "https://login.microsoftonline.com/Enter_the_Tenant_Name_Here"
+
 CLIENT_ID = "Enter_the_Application_Id_here"
-CLIENT_SECRET = "Enter_the_Client_Secret_Here"
-SCOPE = ["https://graph.microsoft.com/User.Read"]
-REDIRECT_URI = "http://localhost:5000/getAToken"
+
+# You can find more Microsoft Graph API endpoints from Graph Explorer
+# https://developer.microsoft.com/en-us/graph/graph-explorer
+ENDPOINT = 'https://graph.microsoft.com/v1.0/me/calendars'
```
Comment (jmprieur) [inline app_config.py]: As discussed, there are tenants where this won't work because tenant admins don't allow users to consent

Comment (rayluo) [reply app_config.py]: Use "https://graph.microsoft.com/v1.0/users" instead. Resolving.

File: app_config.py (hunk 2 above)
Comment (rayluo) [reply app_config.py]: The team decision is to settle on still using the placeholder, with inline comment to recommend more secure ways. Resolving for now.

**Comment by jmprieur** [comment]
wooo! 
good job @rayluo @abhidnya13 !

**Pull request merged**